cityofaustin/atd-moped
# Upgrade to react-map-gl v7

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/9706

_Note: This will not be merged until drawing components are updated to use `@mapbox/mapbox-gl-draw` instead of `react-map-gl-draw`. PR to come, but I wanted to try to split this up as much as possible._

Summary of changes:
1. Upgraded from `react-map-gl ^5.2.10` to `react-map-gl ^7.0.18`
2. Removed `react-map-gl-draw` and `react-map-gl-geocoder` (not compatible with react-map-gl v7)
3. Replaced`react-map-gl-geocoder` with `@mapbox/mapbox-gl-geocoder` (this Mapbox plugin is the way forward 📍 🙌 )
4. Removed the portal that added the speed dial and made it a plain component to reduce the complexity there a bit
5. Updated styling on the map controls to align with Mapbox styling - I can revert this but it was screaming at me to tighten up 😅 and was a small effort (box-shadow, spacing, and borders)

## Testing
**URL to test:** 
https://deploy-preview-753--atd-moped-main.netlify.app/moped/

**Steps to test:**

Devs - please feel free to reach out to me for questions about the code diff. Lots of the code changes came from Prettier formatting so apologies about that.

1. Can switch layers with speed dial
2. Can zoom in/out
3. Can find **Barton Springs Pool** with the geocoder
4. Can add and edit a component that is a line (Bike Lane, etc.)
5. Can add and edit a component that is a point (Bike Box, etc.)
6. Can add and edit a component that is multiple lines (Bike Lane, etc.)
7. Can add and edit a component that is multiple points (Bike Box, etc.)
8. Can hover line segments and see tooltip with street label from CTN layer while selecting
9. Can hover line segments and see tooltip with street label from CTN layer while viewing existing project components
10. Map fits to project components if there are some selected on load in Project Summary and Map tabs
11. Map fits to components when selected for edit
12. Map Features dropdown can be used to show/hide lines and point in both view and edit modes

---
#### Ship list
